Nit: Improve readability and remove comments

kaapstorm opened the pull request:
## Technical Summary

While I was reviewing PR https://github.com/dimagi/commcare-hq/pull/36105 I noticed some lines of my own code that annoyed me.

This tweaks them to improve readability and remove comments.

No change to behavior because the `else` clause in the original is equivalent to `(s in (State.Fail, State.Cancelled) ...)`. Tests will verify this.

**Note:** Base branch is **gh/repeaters/backoff-repeater-ff**

## Feature Flag

`BACKOFF_REPEATERS` and `PROCESS_REPEATERS`

## Safety Assurance

### Safety story

No change to behavior

### Automated test coverage

Yes, the change is covered by tests

### QA Plan

No

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

kaapstorm committed:
Nit: Improve readability and remove comments

kaapstorm committed:
Add test

millerdev commented on corehq/motech/repeaters/tests/test_tasks.py:
```diff
@@ -379,6 +379,25 @@ def test_update_repeater_backs_off_on_failure(
+    @patch('corehq.motech.repeaters.tasks.process_repeater')
+    @patch('corehq.motech.repeaters.tasks.RepeaterLock')
+    @patch('corehq.motech.repeaters.tasks.Repeater.objects.get')
+    def test_update_repeater_does_nothing_on_one_failure(
```
> we shouldn't use that as an indication of remote endpoint health since it reflects that HQ didn't even attempt to send anything

+1 agree

> `[Empty, Empty, Empty, Fail, Empty, Empty]` ... [would] back off right? Which also doesn't seem right.


I think I disagree. If all network requests fail, the simplest policy is to back off, and I think that's reasonable. I don't think it's worth making a special case to handle differently if the number of requests performed is very low (e.g., 1).

kaapstorm committed:
Revert "Add test", reorder checks

This reverts commit 7c25acec45883a6cf25b3431d42865a332df848f.

kaapstorm committed:
Only loop once

I need to stop iterating on interating in this function. Moving on.

But just look at that `for ... else`! How often do you get to use that?

kaapstorm committed:
Add comments

kaapstorm commented on corehq/motech/repeaters/tasks.py:
```diff
@@ -494,21 +494,21 @@ def update_repeater(repeat_record_states, repeater_id, lock_token, more):
+                if state in (State.Fail, State.Cancelled):
+                    remote_is_bad = True
             else:
-                # All the payloads that were sent failed. Try again later.
```
:+1:  2a57625ea82

kaapstorm committed:
Merge branch 'master' into nh/rep/nit
